demo/cloudify_server

```diff
@@ -46,32 +47,39 @@ def select_option(msg, options, default = nil)
   selected
 end
 
+unless REQUIRED_ENV_VARS.all? { Config.public_send(it.downcase) }
+  puts "Please set the following variables in the .env file and re-run docker-compose up:"
+  puts REQUIRED_ENV_VARS.join("\n")
+  exit 1
+end
+
 hostname = get_input("Enter host IP address")
 host_id = get_input("Enter host identifier", "")
 location = select_option("Select provider and location", LOCATIONS, 1)
 
 puts "\n\nCloudifying '#{hostname}' server for '#{LOCATIONS[location]}' \n\n"
 
-strand = Prog::Vm::HostNexus.assemble(
-  hostname,
-  provider_name: location.provider,
-  server_identifier: host_id,
-  location: location.name,
-  default_boot_images: ["ubuntu-jammy"]
-)
-
-puts "Waiting public SSH keys\n\n"
-until (ssh_key = strand.reload.subject.sshable.keys.map(&:public_key).first)
-  sleep 2
+if location.provider != HostProvider::HETZNER_PROVIDER_NAME
+  puts "Currently only Hetzner is supported for this demo. Please use a Hetzner server for cloudification."
+  exit 1
 end
-puts "Add following public SSH key to '/root/.ssh/authorized_keys' on your machine\n\n"
-puts ssh_key
 
-print "\n\nPress enter after you add the above SSH key to your machine\n\n"
-gets.chomp
+strand = if (vmh = VmHost.all.find { it.sshable.host == hostname && it.location == location && it.provider.server_identifier == host_id })
```

Comment:
```suggestion
vm_host_ds = VmHost
  .where(id: Sshable.where(host: hostname).select(:id))
  .where(id: HostProvider.where(server_identifier: host_id).select(:id))
  .where(location_id: location.id)
  
strand = if (vmh = vm_host_ds.first)
```

While it's likely that this script is only run on an empty or near empty database, it's probably best to avoid the 3N+1 queries.

Unrelated to this pull request, currently, this only errors if the combination of sshable hostname, location, and server identifier match. If there is a matching sshable host, but the server identifier is different, or vice versa, or the both match but the location doesn't, should we be creating a new strand for it?  Maybe we should, I'm not sure.